examples/example.cc

```diff
@@ -50,20 +50,23 @@ static const char *Token(const void *node) {
 }
 
 static int ChildrenSize(const void *node) {
-  return ((Node *)node)->children.size();}
+  return static_cast<int>(((Node *)node)->children.size());
```

Comment:
Produces a warning with -Wall because of size being unsigned. We could change the return value but probably that would move the warning to the place where the function is added to the interface.

Reply:
Not if you use `size_t` everywhere I guess, which is probably the best type.

Reply:
Yes, what I mean is that we should also update all the clients, since the type is in node_iface.h, but since I'll have to update them anyway to add the iterators I'll change to size_t if you're Ok with that.